lib/helpers.bash

```diff
@@ -92,762 +96,741 @@ alias reload_completion="$(_make_reload_alias completion completion)"
-            fi
-            _enable-thing $subdirectory $file_type $to_enable $load_priority
-        done
-    else
-        typeset to_enable=$(command ls "${BASH_IT}/$subdirectory/available/"$file_entity.*bash 2>/dev/null | head -1)
-        if [ -z "$to_enable" ]; then
-            printf '%s\n' "sorry, $file_entity does not appear to be an available $file_type."
-            return
-        fi
+		enabled_file="${enabled_file%."${file_type}"*.bash}"
+
+		enabled=
+		_bash-it-component-item-is-enabled "${file_type}" "${enabled_file}" && enabled='x'
+		printf "%-20s %-10s %s\n" "${enabled_file}" "[${enabled:- }]" "$(metafor "about-$file_type" < "$f")"
+	done
+	printf '\n%s\n' "to enable $preposition $file_type, do:"
+	printf '%s\n' "$ bash-it enable $file_type  <$file_type name> [$file_type name]... -or- $ bash-it enable $file_type all"
+	printf '\n%s\n' "to disable $preposition $file_type, do:"
+	printf '%s\n' "$ bash-it disable $file_type <$file_type name> [$file_type name]... -or- $ bash-it disable $file_type all"
+}
+
+function _on-disable-callback() {
+	_about 'Calls the disabled plugin destructor, if present'
+	_param '1: plugin name'
+	_example '$ _on-disable-callback gitstatus'
+	_group 'lib'
+
+	callback="${1}_on_disable"
+	_command_exists "$callback" && "$callback"
+}
+
+function _disable-plugin() {
+	_about 'disables bash_it plugin'
+	_param '1: plugin name'
+	_example '$ disable-plugin rvm'
+	_group 'lib'
+
+	_disable-thing "plugins" "plugin" "$1"
+	_on-disable-callback "$1"
+}
+
+function _disable-alias() {
+	_about 'disables bash_it alias'
+	_param '1: alias name'
+	_example '$ disable-alias git'
+	_group 'lib'
+
+	_disable-thing "aliases" "alias" "$1"
+}
+
+function _disable-completion() {
+	_about 'disables bash_it completion'
+	_param '1: completion name'
+	_example '$ disable-completion git'
+	_group 'lib'
+
+	_disable-thing "completion" "completion" "$1"
+}
+
+function _disable-thing() {
+	_about 'disables a bash_it component'
+	_param '1: subdirectory'
+	_param '2: file_type'
+	_param '3: file_entity'
+	_example '$ _disable-thing "plugins" "plugin" "ssh"'
+
+	local subdirectory="$1"
+	local file_type="$2"
+	local file_entity="$3"
+
+	if [[ -z "$file_entity" ]]; then
+		reference "disable-$file_type"
+		return
+	fi
+
+	local f suffix _bash_it_config_file plugin_global plugin
+	suffix="${subdirectory/plugins/plugin}"
+
+	if [[ "$file_entity" = "all" ]]; then
+		# Disable everything that's using the old structure
+
+		for _bash_it_config_file in "${BASH_IT}/$subdirectory/enabled"/*."${suffix}.bash"; do
+			rm -f "$_bash_it_config_file"
+		done
+
+		for _bash_it_config_file in "${BASH_IT}/enabled"/*".${suffix}.bash"; do
+			# Disable everything in the global "enabled" directory
+			rm -f "$_bash_it_config_file"
+		done
+	else
+		plugin_global="$(command ls $ "${BASH_IT}/enabled"/[0-9]*"${BASH_IT_LOAD_PRIORITY_SEPARATOR}${file_entity}.${suffix}.bash" 2> /dev/null | head -1)"
```

Comment:
I guess the single dollar sign after `ls` is a typo. Same a few lines below.

Reply:
Yes, but I have no idea what or why so I didn't touch it. There's *definitely* another pass of cleanup due on this file!

Reply:
Yeah, it was so weird that I just didn't want to touch it. Anyway, now it's fixed 👍